### PR TITLE
Update AWS account ID used in `update-ami-ids` GHA workflow

### PR DESCRIPTION
**Pull request by reedloden**
PR #36153 updated the AWS account ID used for pulling AMIs, but the GHA workflow was not updated to use this updated AWS account ID.

Depends on gravitational/cloud-terraform#3947.

Ref #34282.

### CODE REVIEW

**Review by camscale — APPROVED** [review]
LGTM, once the cloud-terraform PR merges.

**Review by tcsc — APPROVED** [review]

File: .github/workflows/update-ami-ids.yaml
Comment (fheinecke) [inline .github/workflows/update-ami-ids.yaml:46]: Should this move to an env var in github-terraform instead? Seems kinda odd to hard code a role ARN here.

Comment (reedloden) [reply .github/workflows/update-ami-ids.yaml:46]: I have no objections to that. I am just fixing what is currently here. :-)

**Review by zmb3 — APPROVED** [review]

**Pull request merged**
